### DISCUSSION TITLE
Envelope Icon Request

### DISCUSSION THREAD

**Discussion by HartiganHM** [twilio-labs/paste#1322] [Request: Icon] [answered]
Hello!

I'd like to request an "Envelope" icon. I've provided a few options as I don't have a strong preference on how it looks. I'm open to alternative suggestions as well. 

Please let me know if you need anything else!

- **Mock-up description**: The icon will be used in a CTA for "Open a support request" in the SendGrid Help Center
- **Type of icon**: UI Icon

![image](https://user-images.githubusercontent.com/24458700/112686262-5aa66100-8e3b-11eb-8306-5adcdbc3698d.png)
![image](https://user-images.githubusercontent.com/24458700/112686567-c2f54280-8e3b-11eb-92b4-5ca9437a2d74.png)
![image](https://user-images.githubusercontent.com/24458700/112686599-d2748b80-8e3b-11eb-8dbc-0f7f584c01e9.png)

**Comment by corinne-nagel** [comment]
Sounds good @HartiganHM ! When do you need this icon by?

**Reply by corinne-nagel** [reply]
Added a ticket to our backlog and tagged you: https://issues.corp.twilio.com/browse/DSYS-2729

**Reply by HartiganHM** [reply]
Hey @corinne-nagel! No huge rush. We're presenting final mocks next Monday, but won't need the icon for actual UI implementation for a while.

Thanks for the quick response!

**Comment by SiTaggart** [comment]
If your team has a design partner, there is also a really well trodden self service path that you could follow. https://paste.twilio.design/icons/how-to-add-an-icon

**Reply by HartiganHM** [reply]
@SiTaggart thanks for the heads-up! We don't have an official design partner, but we have a few developers on the team with previous design experience (myself included). Is this something we would be able to handle on our own or is having a dedicated designer partner a requirement?

**Reply by serifluous** [reply]
Hey @HartiganHM ! I can help you out from a design perspective.

Can you share a screenshot showing context of where you'll place the icon? We'll likely need one that represents the same thing in Console, so I want to make sure the icon can be used in both places.

**Reply by serifluous** [reply]
Hey @HartiganHM, checking in—do you still need the envelope icon?

**Reply by HartiganHM** [reply]
Hey, @serifluous , sorry for the delay! I got lost in the shuffle and just got back from vacation. 

Here's a screenshot of where we'll be using it. There are three contact options: "Submit ticket", "Chat", and "Callback". This icon would be for "Submit ticket" (top-most button where the `!` icon is currently). If you have another idea for a better icon, we're open to suggestions 🙂 

<img width="925" alt="Screen Shot 2021-04-09 at 2 02 35 PM" src="https://user-images.githubusercontent.com/24458700/118188478-b0d85f00-b3fd-11eb-8fb4-04c0842197a7.png">

**Reply by SiTaggart** [reply]
Hi @HartiganHM 

Have you done any competitor analysis or prior art checks on iconography used to depict support requests or raising a ticket? I think unless you are actually sending an email the envelope icon might not be the most ideal approach, especially within an email product. Some competitor analysis and prior art might help us here to make an informed decision.

As or the other two icon choices, two points worth keeping in mind:

- The icon chosen for request a callback is actually used for agent experiences in other Twilio products. With that in mind, it might be a better choice for "chat with an agent"
- The Request a callback could easily be a phone icon as a replacement.

**Reply by HartiganHM** [reply]
Thanks for the follow-up @SiTaggart. I'll do some research today and respond with an alternative icon choice. The suggestions about the other icons make sense as well. I'll make sure we get the mocks adjusted.

**Reply by HartiganHM** [reply]
Hey @SiTaggart, here are a few examples of what other companies are using. Atlassian and Zendesk didn't have anything applicable for "ticket", "support",  "request", or "contact" as far as I could tell, but Salesforce had a couple.

**Salesforce (LDS)**

> ![image](https://user-images.githubusercontent.com/24458700/118836567-371ef600-b881-11eb-9a46-463346fe91ad.png)
> "Contact Request"

> ![image](https://user-images.githubusercontent.com/24458700/118836624-47cf6c00-b881-11eb-9f67-f0999beec6d6.png)
> "Submit Ticket"

---

Also, we had two more things we wanted clarity about. We're using the below "ticket" icon for the "Support" section of Console, which we also maintain. Would this be sufficient to cover the "Open Support Request"? We're not locked in on the nomenclature either, so something using "ticket" would be doable.

> ![image](https://user-images.githubusercontent.com/24458700/118837892-7568e500-b882-11eb-818e-8d5193119f46.png)

Additionally, we have this icon for "chat" through the Support on Flex app. Would this be preferred over the headset icon?

> ![image](https://user-images.githubusercontent.com/24458700/118838120-a9440a80-b882-11eb-853c-963db68255b7.png)

**Comment by kate-porter** [comment]
Hey @HartiganHM - @serifluous and I played around with some icons based on your suggestions, and this is what we came up with:
![image](https://user-images.githubusercontent.com/5374045/119061249-29589600-b991-11eb-9fb6-dbab9e99b24b.png)

We use the safety ring for other support actions in Twilio Console right now, so we feel that could be a good fit for submitting a support request. We like your suggestion on the chat bubbles like the ones used in Flex, so we used those for the chat option.

For request a callback, we actually though a calendar icon might be a good fit, because it seemed like there might be some sort of scheduling flow after clicking that action. Does the user go through a scheduling flow if they choose that route? If not, the calendar probably doesn't work - let us know what you think there.

**Reply by HartiganHM** [reply]
Hey, @katieporter, thanks for the help! I like the safety ring a lot. There isn't a scheduling flow for the callback option. It's basically just another way to submit a different type of ticket, but the callback option is reserved for a specific tier of customer support packages. Would a phone/headset icon work better in that case?

I think we're probably good to go with the safety ring and chat bubble icons, though. If those icons can be added, that would be great!

**Comment by SiTaggart** [comment]
Whilst @katieporter is out on PTO I'll jump back on in here.

>  Would a phone/headset icon work better in that case?

I definitely think those would be pretty good options, I think we already have icons to cover them in the system already.

As for the support ring and chat bubbles, we'll add a ticket to the backlog for adding those. What kind of timeline are you in?

**Reply by HartiganHM** [reply]
Great, thanks for the help, @SiTaggart. We're looking to have this feature ready for testing by June 22nd so if we could have the icons some time between **June 14 - 18th**, that would be perfect.

**Reply by SiTaggart** [reply]
Cool, I think @katieporter has created Jira tickets for us to follow up.

**Reply by kate-porter** [reply]
Yes, we have tickets in our current sprint (starting today) to get these icons added. We'll be sure to follow up once they are ready for you!

**Reply by HartiganHM** [reply]
That's great, thanks for the help, @katieporter and @SiTaggart!

**Comment by kate-porter** [comment]
Hey @HartiganHM! Just working on getting the final icons created and wanted to share with you to make sure all looks good from your perspective. Let me know what you think of these!

![image](https://user-images.githubusercontent.com/5374045/120685202-d08f0000-c45c-11eb-9353-a4fb017197a8.png)

**Reply by HartiganHM** [reply]
Wow, ⚡ fast! Let me check with the team for feedback. I'll follow-up by EOD.

**Reply by HartiganHM** [reply]
@katieporter One quick question for you: Will these icons be available across all theme? I assume the answer is "Yes", but we'll be using the SendGrid theme, so I just wanted to double check. Thanks!

**Reply by kate-porter** [reply]
Yep, they'll be available across all themes!

**Reply by HartiganHM** [reply]
Great, then I think we're ready to rock! Thanks for the quick turnaround! 🔥

**Reply by HartiganHM** [reply]
Hey, @katieporter! I just wanted to check in and see what the timeline was for these icons. Our timeline has pushed out a little bit, but it would be great to have them in the next week or so, if possible. Thanks!

**Reply by serifluous** [reply]
Just tagged you in the ticket that's in our current sprint. @andipants12 should be following up soon when it's added!

**Comment by andioneto** [comment, marked as answer]
@HartiganHM Hi there 👋 This was just released and is available from `@twilio-paste/icons` starting with version 5.3.1. The icon component names are `ChatIcon` and `SupportIcon`, and are viewable on the website now. Please take a look and let us know if there is anything else. Thank you!
